### ISSUE TITLE
[Bug]: IconButton wrapped in a parent element with role="grid" accessibility issue.

### ISSUE THREAD

**Issue by abhishek9909** [microsoft/fluentui#23931]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
Chrome
```

### Reproduction

https://codepen.io/Ravitt/pen/wvmMydr?editors=1111

## Actual Behavior
When iconButton wrapped inside a parent element with role="grid" at any depth, the button becomes unresponsive to doubleClicks under talkback mode in android. This is a reference to the codepen link where we repro-ed the issue: https://codepen.io/Ravitt/pen/wvmMydr?editors=1111

## Expected Behavior
The iconButton should be responsive regardless of whether the parent element have role grid or not.

## Possible extension
This behaviour is possibly seen in all buttons, the onclick functionality does not work in talkback when aria-role is set to grid.

### Requested priority

Normal

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by gouttierre** [comment]
Thanks for filing this, but I was unable to reproduce the stated issue. Please, can you check your code-pen you shared with us as it does not seem to work on my end and let us know if the issue still persists with an updated code-pen. Thank you in advance.
